### DISCUSSION TITLE
GoPro Hero 12 Labs

### DISCUSSION THREAD

**Discussion by DemonViewLLC** [gopro/labs#513] [General]
The GoPro Labs firmware is something everyone should be using (imho).  Can anyone say if the GoPro Labs Firmware will be available on the Hero 12 on the release date or if it will be a few weeks later (which seems to be the standard).  I realize there may be an embargo on information until after the official release date, however it would be good to know if I can still recommend installing this firmware to new owners so they can enter the owner information into the camera.

**Comment by dnewman-gpsw** [comment]
Working for a synchronous release or as near to as possible.  Same day or within a few, not two weeks.

**Reply by ToddKortte** [reply]
When will Hero12 options be available in the QR code creator?

**Reply by dnewman-gpsw** [reply]
https://gopro.github.io/labs/control/custom/ is open source, anyone can contribute. I do these updates typically, but lots of other Labs work is taking priority.

**Comment by Deadcode85** [comment]
I have a few question about the Hero 12 readout modes:

1., In 5.3K / 4K60 LTC gave awesome dynamic range on the Hero 11. It was DOL-HDR readout?

2., Now with the Hero 12 in these resolution/framerates it's advertised as dual readout in HDR mode. Does the extended dynamic range available in GP-LOG mode too? I mean: in log it clips highlights sooner in 5.3K 60fps compared to 5.3K 30fps? Or sensor dual readout is not available while shooting log?

3., How does the GP-Log differs from LOGB=400 at Hero 11?

4., Does all resolution and framerates have 10 bit recording now? For example 2.7K 4:3 60 fps?

5., Will the other (4K/5.3K) 4:3 resolution added back in later firmware?

**Reply by dnewman-gpsw** [reply]
1. HERO11 only used DOL-HDR in photos.

Tech details beyond the published specs still has to wait a few days.   I think Monday or Tuesday all the embargos lift.

**Reply by Deadcode85** [reply]
I played a little with some footage from DC Rainmaker:
2., No, HDR readout is only available in HDR mode, no extended DR sensor readout in GP-LOG. Shame :)
3. you already answered it in another topic, LOGB400 and GP-LOG is the same, but it seems like the NR is on (or not 1) by default.

**Reply by dnewman-gpsw** [reply]
3. While HERO12 does use the Labs NR01, it does tune the noise reduction to be more Log friendly. Labs on HERO12 will add full NR01 control.
4. 2.7K is still 8-bit.
5. No idea.

**Reply by dnewman-gpsw** [reply]
but case is important, mL

**Comment by kouyoumin** [comment]
> I played a little with some footage from DC Rainmaker: 2., No, HDR readout is only available in HDR mode, no extended DR sensor readout in GP-LOG. Shame :) 

What if we set LOGB=400 (using QR control) when shooting in HDR mode?

**Reply by dnewman-gpsw** [reply]
I tried that, doesn't work like you might hope.

**Reply by Deadcode85** [reply]
what if you turn off tonemapping for HDR, then LOGB400?

**Reply by dnewman-gpsw** [reply]
I just tried, mixed results. Potentially it is be using one of the exposures with tone-mapping off, but I'm not sure. The image is much brighter, so more shadows, but clipping highlights, explaining my theory.

**Reply by Hydrostreamin** [reply]
@dnewman-gpsw awesome. I’ll play around with it some more. Thank you sir!

**Comment by dnewman-gpsw** [comment]
Labs for HERO12 is out.

**Reply by DemonViewLLC** [reply]
Excellent!  Thank you for letting us know!

**Reply by ToddKortte** [reply]
The firmware is out, but the code generator is not updated for the Hero12.  When can we generate codes with HDR and LOG options?

**Reply by dnewman-gpsw** [reply]
Just type mL - mode Log or mH - mode HDR, into the Addition Command field.

**Reply by Hydrostreamin** [reply]
@dnewman-gpsw 

Like this? 

![IMG_0084](https://github.com/gopro/labs/assets/145780373/fb66fd85-3c56-4c64-8270-24a814b5024e)

**Comment by markusfinholt** [comment]
I will be making an informative video on gopro log and all of its confusions, plus how to grade with a display or a scene referred workflow. I'd love some answers so I don't spew any misinformation.

- Will GP-Log / LOGB / WIDE / NR01 be available with the Hero 12 and the Max Lens Mod 2.0?

- Is GP-Log only a LOGB=400 transform, or does it include a WIDE=1 to rec709 transform as well?

- It says on Gopro's sites that GP-Log is limited to ISO 400. is this the lower limit, or the upper limit?

- Is there any information anywhere on the gamut size of WIDE=1?

- Is the GP-Log to Rec709 LUT the same as the IDT / ODT from this user? https://github.com/xtremestuff/protune-transforms

I kind of need these answers before I purchase the Hero 12. Thanks as always for the great work.

**Reply by dnewman-gpsw** [reply]
1) Labs is available now for HERO12.  GP-Log in not offered with MLM2.0 modes, but LOGB still works, so I expect this was to reduce mode permutations on testing. 
2) GP-Log is LOGB=400 at EV comp of -2, with some customized tuning. Using Labs you can combine NR01 and WIDE with the GP-Log mode.  So shooting Labs with cFx-2i4oMLOGB=400 is very similar to GP-Log.
3) 400 is ISO Max setting, ISO Min is at 100, so you can auto range from 100-400, or reduce ISO Max if you wish.
4) Maybe from Sony, it is the native colors from the Sony sensor.
5) No idea, I haven't tried those.

You can't make "an informative video" very well without a HERO12, so I hope these answer your questions enough to get one.

**Reply by markusfinholt** [reply]
Thanks. I was not planning on making a video without the Gopro 12, it was more about it being worth the purchase for me. If LOGB and GP-Log was not possible with the MLM2.0 I would never purchase the 12, since I already have the 11.

You say GP-Log is LOGB=400 at EV-2. Interesting choice when the mid gray value is at 68% IRE. I would think that you would get a very dark image with crushed shadows when setting it so low. I have found some very good results with LOGB=400 at ev comp 0 and damp=20. ( https://youtu.be/_V_M-xNajgA?si=zry1AGjCpFXLq055&t=256 )

What's the decision behind setting the EV comp to -2?

Is the EV comp locked when selecting GP-Log?

**Reply by dnewman-gpsw** [reply]
EV Comp is locked with GP-Log.  It wasn't my decision for use -2 with Log 400.  But I also wouldn't lefted it a 0 EV. In my FPV shooting I use LOGB=400 ev-1, even with Flat I use EV -0.5.  I expect the decision was to go for more headroom with a look that works okay without color correction (but still needs it). My guess. But it looks okay, and shadows not too crushed.  

This compare the curves.  LOG 400 at EV-2 is still brighter than a standard Rec709 transfer function for all the useable range. https://www.desmos.com/calculator/emvi0n5rug

**Reply by p0las** [reply]
>Is the GP-Log to Rec709 LUT the same as the IDT / ODT from this user? https://github.com/xtremestuff/protune-transforms
the transforms will not work as they are for gamma curve LOGB=113 not 400.
I will make one for you. I had one and tried LOGB=400 but didn't see much improvement over the flat (113). 

EV -2 for  gp-log seems a bit aggressive way to preserve highlights. since this is the same sensor as gopro 11 I would expect a fair bit of noise. everytime I pushed 11 that much the noise was out of control and the noise reduction was messing with the image way too much. I would love to see some actual footage from 12.

**Reply by markusfinholt** [reply]
Let me know when you've made the IDT to Davinci Wide Gamut with WIDE=1 and LOGB=400. I'll buy the Gopro 12 tomorrow and can send you some footage. Don't have a color checker though.

Am I wrong in thinking adding contrast would suffice since the log base isn't too different between LOGB=116(flat) and LOGB=400? I simply added contrast in my videos and got a near 1 to 1 match with my cine cameras.

**Reply by dnewman-gpsw** [reply]
I agree it is aggressive, not may decision, just providing information.

**Reply by dnewman-gpsw** [reply]
Here is Natural 4K
![image](https://github.com/gopro/labs/assets/10856084/54dd2ddd-7cd8-4283-b17a-3d69f68d9ed5)

**Reply by dnewman-gpsw** [reply]
Here is Flat 4K
![image](https://github.com/gopro/labs/assets/10856084/c06b885d-5161-472e-9a56-614d501307ba)

**Reply by dnewman-gpsw** [reply]
Here is GP-Log 4K
![image](https://github.com/gopro/labs/assets/10856084/b59610dc-aaa7-4d1d-bb90-3fb1d595c21d)

**Reply by dnewman-gpsw** [reply]
GP-Log has the headroom, and enough enhancement in the curve to present -2EV as a nice image that is simple to color correct. I do believe  using Labs and NR01 will offer more in the shadows for those will to do post noise reduction when needed.

**Reply by p0las** [reply]
gopro 11 with LOGB=400 underexposed by 2 stops:
https://nopixels.net/articles/unhappy-with-gopro-11-footage-try-this/#log
it is definitely worse in shadows than underexposed by -0.5. hero 12 must have some other changes to compensate for that.

**Reply by dnewman-gpsw** [reply]
I don't think so, it is just whether you favor shadows or highlights.

**Reply by RikshaDriver** [reply]
With the new HERO12 firmware, is GP-Log now set at 0EV?

On a side note, the new non-labs firmware has a bug as it's unable to read the timecode QR code.

**Reply by dnewman-gpsw** [reply]
Yes GP-Log now defaults to 0EV, and with the EV control restored.  

In more testing timecode QR Code is working on 1.20.

Labs versions will be out today.

**Reply by RikshaDriver** [reply]
> Yes GP-Log now defaults to 0EV, and with the EV control restored.  
> 
> 
> 
> In more testing timecode QR Code is working on 1.20.
> 
> 
> 
> Labs versions will be out today.
> 
> 

Thanks David,

So this is now effectively LOGB=400 at 0EV which is back to SDR range with the same clipping point as Protune Flat. Bit of a shame as Protune Flat offers more unique bits in the encoding.

**Reply by dnewman-gpsw** [reply]
Dynamic range hasn't changed. EV is just unlocked, set it to -2 is you want more headroom, or 0EV for more shadow detail.  Flat (log113) and no more "unique" bits than Log400.

**Reply by RikshaDriver** [reply]
Understand DR is the same, however the range allocated to highlights is now reduced to peak at ~2.5 stops. The advantage pre-1.20 f/w had was that this -2EV exposure offset was baked in from the onset - what should have been fixed with this was a +1EV bias correction for exposure metering instead. I will still use -1 or -2 EV with Labs, but there is no difference between Flat and the updated GP-Log as far as the stock firmware is concerned.

Some time ago, I had calculated 10-bit linear input (0-1023) to Flat & GP-Log/LOGB400 outputs respectively and hence the assertion that Flat offers more unique code values (I mentioned bits in haste but was referring to the bit code values): https://www.desmos.com/calculator/smt1hcbs11

From my calculations, Flat provides 546 unique code values whereas LOGB400 provides 475 unique code values... unless the OETF is down sampled from 12-bit linear Input to 10-bit Output or some other calculation method is used, in which case I'm completely wrong.

**Reply by dnewman-gpsw** [reply]
The same math argument says we get more codewords encoding linear light, and from experience I know how bad an idea that is.  Curve emphasis helps with compression and noise in the system.  There only linear light encoded in uncompressed systems.  Yes, you are also assuming the image pipeline is 10bit, when it is deeper than that.

**Comment by mariustoicescu** [comment]
Hello, can it be used with GoPro 12 + Max Wide Lens Mod 2.0?

**Reply by dnewman-gpsw** [reply]
What can be used? Labs support all modes on all cameras since HERO7.

**Reply by mariustoicescu** [reply]
:)) this is your best answer. Thx for all your help!

**Comment by NilsVega** [comment]
Hello, can anyone tell me whether gopro labs is better suited for gopro 11 than for gopro 12. It seems to me that several functions of Labs do not work correctly on gopro 12

**Reply by dnewman-gpsw** [reply]
HERO12 is better as it is newer, except for GPS functions for the H11.  Labs gets more updates on the newer cameras.  Labs for the HERO13 is best yet.

**Reply by NilsVega** [reply]
I correctly understood that gopro labs is better for H.11 than H,12

**Reply by dnewman-gpsw** [reply]
The opposite. HERO12 is better as it is newer, and it has significant robustness improvements (script reliability.) If you need GPS, then definitely get the HERO13, better still.  Only reason to get HERO11 today is for the reduced price, but H12 and H13 are the better cameras.
